lib/util/itinerary.js

```diff
@@ -15,3 +15,45 @@ export function getLeafletLegBounds (leg) {
 export function isBatchRoutingEnabled (otpConfig) {
   return Boolean(otpConfig.routingTypes.find(t => t.key === 'BATCH'))
 }
+
+// TODO: move to OTP-UI + add tests?
+/**
+ * @returns true if at least one of the legs of the specified itinerary is a transit leg.
+ */
+export function itineraryHasTransit (itinerary) {
+  if (itinerary && itinerary.legs) {
+    for (const leg of itinerary.legs) {
+      if (coreUtils.itinerary.isTransit(leg.mode)) {
+        return true
+      }
+    }
+  }
+
+  return false
+}
+
+// TODO: move to OTP-UI + add tests?
+/**
+ * @returns true if at least one of the legs of the specified itinerary is a rental leg
+ *   (e.g. CAR_RENT, BICYCLE_RENT, MICROMOBILITY_RENT).
+ */
+export function itineraryHasRental (itinerary) {
+  if (itinerary && itinerary.legs) {
+    for (const leg of itinerary.legs) {
+      if (leg.mode.indexOf('_RENT') > -1) {
+        return true
+      }
+    }
+  }
+
+  return false
+}
+
+/**
+ * Determines whether an itinerary can be monitored.
+ * @returns true if at least one of the legs of the specified itinerary is
+ *   a transit leg, and none of the legs is a rental leg (e.g. CAR_RENT, BICYCLE_RENT, etc.).
+ */
+export function itineraryCanBeMonitored (itinerary) {
+  return itineraryHasTransit(itinerary) && !itineraryHasRental(itinerary)
+}
```

Comment:
These 3 functions could probably be combined into one so that a single loop occurs which checks each leg as needed. Furthermore, the `coreUtils.itinerary.hasRental` could be used. Also, this must check if a ridehail leg exists (`coreUtils.itinerary.hasHail` can be used). Also, I noticed that `lib/util/state.js` has a duplicate function of `itineraryHasTransit`. This redundancy should be avoided. Regarding testing, I think making tests for the `itineraryCanBeMonitored` would be good.

Reply:
Yeah, I totally missed on the duplicate in `state.js` but that function is no longer used for the purpose of this PR. The changes are in https://github.com/opentripplanner/otp-react-redux/pull/266/commits/df6bf8da862d7ac1bee6f28cdb581c5cb581cb8d, and the tests in https://github.com/opentripplanner/otp-react-redux/pull/266/commits/aa19444390bafc983ddc5ed51c0eb6fbd0aab1ae.